src/browser/components/LoadingAnimation/LoadingIcon.jsx

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+function LoadingAnimation() {
+  return (
```

Comment:
I'd actually rather have the SVG be in a `.svg` file and inlined during build but didn't have time to implement that given the infrastructural changes needed. Instead I opted for a separate private component because the SVG is large and cumbersome and for me makes this component more difficult to understand when combined. Since the control of the SVG is handled almost exclusively by CSS with some basic JS state changes (via class names), I opted for keeping the SVG entirely separate. The index.js file exposes this file as the public representation of the component allowing the SVG to remain private.